server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java

```diff
@@ -505,14 +513,17 @@ private void addAssignedToMeFacetIfNeeded(SearchRequestBuilder builder, SearchOp
     String facetName = IssueFilterParameters.FACET_ASSIGNED_TO_ME;
 
     // Same as in super.stickyFacetBuilder
-    Map<String, FilterBuilder> assigneeFilters = Maps.newHashMap(filters);
-    StickyFacetBuilder assignedToMeFacetBuilder = new StickyFacetBuilder(queryBuilder, assigneeFilters);
-    BoolFilterBuilder facetFilter = assignedToMeFacetBuilder.getStickyFacetFilter("__assigned_to_me");
+    StickyFacetBuilder assignedToMeFacetBuilder = new StickyFacetBuilder(queryBuilder, filters);
+    BoolFilterBuilder facetFilter = assignedToMeFacetBuilder.getStickyFacetFilter("__isAssigned", fieldName);
 
-    builder.addAggregation(AggregationBuilders
-      .filter(facetName)
+    FilterAggregationBuilder facetTopAggregation = AggregationBuilders
+      .filter(facetName + "__filter")
       .filter(facetFilter)
-      .subAggregation(AggregationBuilders.terms(facetName + "__terms").field(fieldName).include(login)));
+      .subAggregation(AggregationBuilders.terms(facetName + "__terms").field(fieldName).include(login));
+
+    builder.addAggregation(
+      AggregationBuilders.global(facetName)
+        .subAggregation(facetTopAggregation));
```

Comment:
Wrapping the ``filter`` aggregation in a ``global`` aggregation, to avoid having the query filter applied to the aggregation.